xarray/core/indexing.py

```diff
@@ -164,7 +95,20 @@ def get_loc(index, label, method=None, tolerance=None):
 
 
 def get_indexer(index, labels, method=None, tolerance=None):
+    """ Call pd.Index.get_indexer(labels). If labels are Variable,
+    The return type is also a Variable with the same dimension to
+    labels.
+    """
+    from .variable import Variable
+
     kwargs = _index_method_kwargs(method, tolerance)
+    if isinstance(labels, Variable):
+        if labels.ndim > 1:
+            indexers = np.array(index.get_indexer(labels.data.flatten(),
+                                                  **kwargs))
+            return Variable(labels.dims, indexers.reshape(labels.shape))
+        else:
+            return Variable(labels.dims, index.get_indexer(labels, **kwargs))
```

Comment:
I prefer to write this sort of logic with as few branches as possible, e.g.,
```python
flat_labels = np.ravel(labels)
flat_indexer = index.get_indexer(flat_labels, **kwargs)
indexer = flat_indexer.shape(labels.shape)
if isinstance(labels, Variable):
    indexer = Variable(labels.dims, indexer)
return indexer
```

Reply:
Done